### DISCUSSION TITLE
Centralized logging system for Mobile apps

### DISCUSSION THREAD

**Discussion by Abhijith-net** [MobileNativeFoundation/discussions#13] [Proposals]
Centralized logging helps solve production incidents better. 

### Proposal
1. Logging based on taxonomy
2. Taxonomy which can stitch log of mobile apps with the backend services. eg: ORDER_ID being the same key across Mobile Apps and Backend services help understand the flow of Order placement till delivery!
3. Need to distinguish sessions. Sessions can be based on app launch / pause / resume
4. Can have multiple instances with configuration to support different destinations to upload to. This implies, the same SDK to be supported for various cases like Debugging and Analytics!
5. Pluggable filters and Enrichers

### Reference
https://blogs.halodoc.io/transporter-centralised-logging-in-apps/

**Comment by shepting** [comment]
We would love to have something based on `os_log` but Apple seems really uninterested in making that work well.

**Reply by Abhijith-net** [reply]
Custom logs should still workout!

**Reply by sachinvas-halodoc** [reply]
@shepting Yes, `os_log` is good however lacks the flexibility of on-demand logs upload and the user has to share these with developers.

On the other hand, `Transporter` would resolve these drawbacks and keeps it consistent across mobile platforms.

**Reply by steipete** [reply]
http://doesoslogstoreworkyet.com/ 🙃

**Comment by freak4pc** [comment]
I've been really interested in finding a great solution to this and surprisingly every product I find has serious drawbacks. Either very wasteful networking, or bad UI, or bad filtering, etc. 

Would be interested to learn what other folks are using to solve this, too.
